### PR TITLE
Issue #31: Support newtype structs

### PR DESCRIPTION
**Pull request by magnet**
This PR fixes the handling of newtype structs.

## What did you implement:

I don't forward newtype_struct to the any dummy serializer, instead implement the method manually in the `Val` and `VarName` deserializers (I can't see why `VarName` would require it, but it won't hurt!).

Closes: #34

#### How did you verify your change:

I added a custom newtype struct to the tests, and added such a field to the current tested struct (`Foo`). It works :smile: 

#### What (if anything) would need to be called out in the CHANGELOG for the next release:

This is a behavior that people would expect -- but it might be good to say Envy now supports `newtype` structs.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -457,6 +488,7 @@ mod tests {
                     debug_mode: false,
                     size: Size::Small,
                     provided: Some(String::from("test")),
+                    newtype: CustomNewType(String::from("42"))
```
Comment (softprops) [inline src/lib.rs]: fancy!

Comment (magnet) [reply src/lib.rs]: I just made it fancier by *not* using a `String` in the newtype, so we now test that the inner time gets properly serialized as well.

**Comment by magnet** [comment]
(Looks like I'm tired, aside from random typos I set the wrong issue number. Fixed)

**Comment by magnet** [comment]
So that build failed because nightly is slightly broken (rustfmt is missing). May I suggest not letting nightly fail builds?  Something as the following:
```yaml
matrix:
  allow_failures:
    - rust: nightly```

**Comment by softprops** [comment]
that works. looks like a problem with the rustfmt project. I'll follow up with them

**Comment by magnet** [comment]
@softprops I'm usually working on nightly and these breakages happen a lot due to upstream backward incompatible changes between rustc, std, etc. I'll submit a PR to update your Travis to make failing builds when nightly builds fail, and you decide if you want it, but in my experience it's a good setting to have.

**Comment by softprops** [comment]
published in 0.4.0 https://crates.io/crates/envy

**Pull request merged**